kubernetes/llama-stack/README.md

```diff
@@ -0,0 +1,141 @@
+podman push quay.io/<quay user name or org/<image name>
+```
+
+Update the [`deployment.yaml`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/deployment.yaml#L71) file using the image generated above.
+
+## Configmap
+
+A predefined [`configmap`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/configmap.yaml) exists with entirely too many options, but is a placeholder for what can be added. Create the configmap defining your unique settings. Then create the CM. The [`template.yaml`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/kubernetes/llama-stack/template.yaml) file does not need to be modified. It contains the jinja chat template.
+
+```
+oc create -f llama-stack/configmap.yaml
+oc create -f llama-stack/template.yaml
+```
+
+## Deployment
+
+To create the Llamastack deployment:
+
+```
+oc create -f llama-stack/deployment.yaml
+```
+
+## Access
+
+This deployment isn't exposed so depending on where you need to access it from outside the cluster, expose it accordingly.
+
+If you want to deploy a chat server that points to the llamastack deployment then run the following:
+
+```
+oc expose deployment/llamastack-deployment
+```
+
+If you want to access the llama-stack externally from the cluster then you must expose the svc which will create a route:
+
+```
+oc expose svc/llamastack-deployment
+```
+
+The above will generate a route similar to this:
+
+```
+llamastack-deployment-llama-serve.apps.ocp-beta-test.nerc.mghpcc.org
+```
+(_**NOTE:**_ This will be unique to your environment)
+
+## Using the endpoints
+
+Now that everything is available in the OpenShift cluster, you can utilize the vLLM instance and the Llamastack server by ensuring that you set the following before running your tests/experiments:
+
+```
+export INFERENCE_MODEL="meta-llama/Llama-3.2-1B-Instruct"
+export LLAMA_STACK_PORT=80
+export INFERENCE_ADDR=vllm-llama-serve.apps.ocp-beta-test.nerc.mghpcc.org
+```
+
+(_**NOTE:**_ The INFERENCE_ADDR will be based on your OpenShift environment. The INFERENCE_MODEL will be the model you have deployed via vLLM)
+
+## Testing the endpoints
+
+To test the Llamstack server, you can run any of the examples from [here](https://github.com/opendatahub-io/llama-stack-demos/tree/main/demos/rag_agentic/src). Make sure to set the following env vars before executing the scripts:
+
+```
+LLAMA_STACK_PORT=80
+INFERENCE_MODEL="meta-llama/Llama-3.2-1B-Instruct"
+```
+
+You will also need to update the script with the correct connection settings for the deployed Llamstack server URL. For example, if you wish to run the [`0_simple_agent.py`](https://github.com/opendatahub-io/llama-stack-demos/blob/main/demos/rag_agentic/src/0_simple_agent.py) script, you will need to update [this line](https://github.com/opendatahub-io/llama-stack-demos/blob/main/demos/rag_agentic/src/0_simple_agent.py#L34) with the URL of your deployed Llamastack server.
```

Comment:
The notebooks should be pulling from the `.env` file. Users should not need to update any part of the notebooks.